### PR TITLE
[Feature] - Added ZEN module for creating loadout lockers mid-mission.

### PR DESCRIPTION
**Pull request by Bubbus**
### Pull Request Description
**When merged this pull request will:**
- Add ZEN module for creating loadout lockers mid-mission.
- Make loadout locker look better if a different object is used (the locker is technically from the 3den objects mod so this adds some really minor and petty vanilla compatibility).

### Release Notes
Added ZEN module for creating loadout lockers mid-mission.

## IMPORTANT

- [x] Testing has been completed as neccessary, depending on the nature & impact of the changes.
- [x] The Release Notes section below must be filled out appropriately to explain the changes made in this PR. This section will be used in Framework Changelogs.
- [x] If the contribution affects [the wiki](https://github.com/CombinedArmsGaming/CAFE3/wiki), please include your changes in this pull request so the wiki is consistently updated.
- [x] [Contribution Guidelines](https://github.com/CombinedArmsGaming/CAFE3/blob/release/contributing.md) are read, understood and applied.
- [x] Title of this PR uses our standard template `[Descriptor] - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: components/gearScript/zen/fn_zen_createLoadoutLocker.sqf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include "../macros.hpp"
+
+CLIENT_ONLY;
+
+params ["_position", "_unit"];
+
+if ((!isNull _unit) and {_unit isKindOf "CAManBase"}) exitWith
+{
+    ["Cannot use this module on units.  Try on ground, vehicles or containers."] call zen_common_fnc_showMessage;
+};
```
Comment (Cre8or) [inline components/gearScript/zen/fn_zen_createLoadoutLocker.sqf]: What happens if the function is called on, say, an object (e.g. by placing the Zeus module on a vehicle)? This condition won't catch that, so the function continues to run, causing the dialog to pop up. Not critical, but... odd? Is this intended?

File: components/gearScript/zen/fn_zen_createLoadoutLocker.sqf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+private _createLocker = 
+{
+    params ["_dialogValues", "_args"];
+    _dialogValues params ["_chosenFaction"];
+    _args params ["_position", "_unit"];
```
Comment (Cre8or) [inline components/gearScript/zen/fn_zen_createLoadoutLocker.sqf]: `_unit` is not referenced inside `_createLocker`. Continuing from my comment above: if executed on a vehicle, the dialog shows up. Zeus confirms their options and confirms the dialog, causing `_createLocker` to run. Next thing they know, a locker is spawned inside of the vehicle - and ArmA™ ensues.

Am I missing something?

Comment (Joecuronium) [reply components/gearScript/zen/fn_zen_createLoadoutLocker.sqf]: Agree with cre here, the fact that the locker always seems to be spawned even though the exitWith statement would suggest the user its usable on vics and containers seems contradictory

**Review by Joecuronium — DISMISSED** [review]
Clears it up. Greenlight from my end.

**Comment by Cre8or** [comment]
I take it assigning existing vehicles/objects as lockers is not required, then? In that case, greenlight from me too.

**Comment by Bubbus** [comment]
Turning an MRAP into a loadout locker could be funny.  Should we?

**Comment by Cre8or** [comment]
I dunno, *should we?*

It certainly sounds like a useful feature to have, in line with the ability to turn objects/vehicles into ACE arsenals.

**Comment by dgibso29** [comment]
I'm in favour of loadout vehicles, that seems potentially handy

**Review by Cre8or — COMMENTED** [review]
Neat.

🚢

**Review by Cre8or — APPROVED** [review]

**Pull request merged**